common/persistence/sql/execution_tasks.go

```diff
@@ -411,8 +393,8 @@ func (m *sqlExecutionStore) populateGetReplicationTasksResponse(
 	}
 	var nextPageToken []byte
 	lastTaskID := rows[len(rows)-1].TaskID
-	if lastTaskID < requestMaxReadLevel {
-		nextPageToken = serializePageToken(lastTaskID)
+	if lastTaskID+1 < exclusiveMaxTaskID {
+		nextPageToken = serializePageToken(lastTaskID + 1)
 	}
```

Comment:
maybe make a util func for this:
func nextPageToken(lastTaskID, exclusiveMaxTaskID int64) []byte {
...
}

Reply:
Created `getImmediateTaskNextPageToken` and renamed `getTaskRange` to `getImmediateTaskReadRange`, so that they won't be confused with the logic for timer/scheduled tasks.